protocol/vm/context.go

```diff
@@ -0,0 +1,31 @@
+package vm
+
+// Context contains the execution context for the virtual machine.
+//
+// By convention, variables of this type have the name context, _not_
+// ctx (to avoid confusion with context.Context).
+type Context struct {
+	VMVersion uint64
+	Code      []byte
+	Arguments [][]byte
+
+	TxVersion *uint64
+
+	BlockHash            *[]byte
+	BlockTimeMS          *uint64
+	NextConsensusProgram *[]byte
+
+	TxSigHash        *[]byte
```

Comment:
The slices that are hashes could be `nil` to mean "absent," because we know they should be non-empty if present. But other slices in this struct could be "empty and present," so those do need to be pointers to slices. My (weak) aesthetic preference was to handle all the slices that way.